eclass/xdg.eclass

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+# @ECLASS: xdg.eclass
+# @MAINTAINER:
+# freedesktop-bugs@gentoo.org
+# @AUTHOR:
+# Original author: Gilles Dartiguelongue <eva@gentoo.org>
+# @BLURB: Provides phases for XDG compliant packages.
+# @DESCRIPTION:
+# Utility eclass to update the desktop and shared mime info as laid
+# out in the freedesktop specs & implementations
+
+inherit xdg-utils
+
+case "${EAPI:-0}" in
+	4|5|6)
+		EXPORT_FUNCTIONS src_prepare pkg_preinst pkg_postinst pkg_postrm
+		;;
+	*) die "EAPI=${EAPI} is not supported" ;;
+esac
+
+DEPEND="
+	dev-util/desktop-file-utils
+	x11-misc/shared-mime-info
+"
+
+# @FUNCTION: xdg_src_prepare
+# @DESCRIPTION:
+# Prepare sources to work with XDG standards.
+xdg_src_prepare() {
+	xdg_environment_reset
```

Comment:
Actually, creation of those directories only makes sense for actual build operation, if you are installing a binary package for example, it will not be useful to have directories exist and variables set.